src/Tools/BuildValidator/BuildConstructor.cs

```diff
@@ -59,53 +62,119 @@ public Compilation CreateCompilation(MetadataReader metadataReader, string name)
             throw new InvalidDataException("Did not find language in compilation options");
         }
 
-        private ImmutableArray<MetadataReference> CreateMetadataReferences(CompilationOptionsReader pdbReader)
+        private ImmutableArray<MetadataReferenceInfo> GetMetadataReferenceInfos(CompilationOptionsReader compilationOptionsReader)
+        {
+            var referenceInfos = compilationOptionsReader.GetMetadataReferences();
+            return referenceInfos;
+        }
+
+        private ImmutableArray<SourceFileInfo> GetSourceFileInfos(CompilationOptionsReader compilationOptionsReader, Encoding encoding)
+        {
+            return compilationOptionsReader.GetSourceFileInfos(encoding);
+        }
+
+        private void LogResolvedMetadataReferences(ImmutableArray<MetadataReferenceInfo> infos, ImmutableArray<MetadataReference> references)
```

Comment:
This could be a local function and would remove the need to have the explicit arg checking, could be an assert instead.

Reply:
Fixed